t/unit/test_transport.py

```diff
@@ -700,7 +700,22 @@ def test_wrap_socket_ciphers(self):
             set_ciphers_method_mock.assert_called_with(sentinel.CIPHERS)
 
     def test_wrap_socket_sni_cert_reqs(self):
-        # testing _wrap_socket_sni() with parameter cert_reqs == ssl.CERT_NONE
+        # testing context creation inside _wrap_socket_sni() with parameter 
+        # cert_reqs == ssl.CERT_NONE
+        with patch('ssl.SSLContext.wrap_socket') as mock_wrap_socket:
+            sock = Mock()
+            self.t._wrap_socket_sni(
```

Comment:
Usually, we don't test the internals of a class.
We exercise the public API and check the outcomes.
This is because this test may pass but another code path in the public method will still make the desired functionality unobtainable.